### PR TITLE
docs: provide initial docsify documentation

### PR DESCRIPTION
**Pull request by Alex-Werner**
## Issue being fixed or feature implemented

This PR brings docsify documentation for our repository.

## What was done?
- feat: provide docsify unpkg embedded library for github pages
- docs: DataContract, Identity and Document primitives
- docs: dpp.dataContract usage, dpp.document usage and dpp.identity usage

## How Has This Been Tested?
N/A

## Breaking Changes
N/A

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [X] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by Alex-Werner** [comment]
This is just an initial draft that covers Document, DataContract and Identity primitives and facade usage. 
The decision to cover theses first is due to get the ball rolling on our documentation here. 
But more work would be needed which we should plan first on our sprint.

File: docs/index.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        '/': 'No results!'
+      },
+      paths: 'auto',
+      placeholder: {
+        '/': 'Search'
+      }
+    },
+    // search: 'auto',
+    formatUpdated: '{MM}/{DD} {HH}:{mm}',
+    themeColor: '#008de4',
```
Comment (thephez) [inline docs/index.html:35]: I like to include this so it doesn't drop you in the middle of a page when you change pages.
```suggestion
    themeColor: '#008de4',
    auto2top: true,
```

File: docs/_sidebar.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- Usage
+    - [new DashPlatformProtocol()](usage/DashPlatformProtocol.md)
+    - [dpp.dataContract](usage/dataContract.md)
+    - [dpp.document](usage/document.md)
+    - [dpp.identity](usage/identity.md)
```
Comment (thephez) [inline docs/_sidebar.md:5]: This link doesn't work due to incorrect file name (link to identity also not working other places for the same reason). There seem to be some inconsistent use of case in file names. Personally I would prefer they start with lowercase, but as long as they're consistent we should be okay.

**Review by thephez — COMMENTED** [review]
@Alex-Werner Took a quick look and spotted these. Didn't do a thorough review as #240 is massive and changes a lot of stuff.

File: docs/_sidebar.md (hunk 2 above)
Comment (Alex-Werner) [reply docs/_sidebar.md:5]: Any incorrect file name is a bug, so I'm double checking this. 
However, reasoning behind lowercase vs uppercase : upperCase (Identity, DashPlatformProtocol) refers to class we can instantiates, while lowercase are namespace. 

When you do 

```
const dpp = new DashPlatformProtocol()
``` 
You got facades that are accessibles : `dpp.dataContract` which are not `DataContract` (defined in primitives/DataContract). 
That is what I'm trying to get explicitly here. How should I fix that ?

File: docs/index.html (hunk 1 above)
Comment (Alex-Werner) [reply docs/index.html:35]: Neat !

**Comment by Alex-Werner** [comment]
Rebased onto 0.16.

File: docs/_sidebar.md (hunk 2 above)
Comment (thephez) [reply docs/_sidebar.md:5]: Hmm, apparently it is may be more complicated than just a broken link. :slightly_smiling_face: I'd say stick with your plan. For this specific case, the identity usage doc is named `docs/usage/Identity.md` while the link is trying to use `docs/usage/identity.md`. If I'm understanding you, the file needs to be renamed to `docs/usage/identity.md`, correct?

**Comment by Alex-Werner** [comment]
https://github.com/dashevo/js-dpp/pull/240 was merged, I've update this PR to represent the new state of master.

File: docs/usage/stateTransition.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+| parameters                   | type                                       | required | Description                                             |  
+|------------------------------|--------------------------------------------|----------| --------------------------------------------------------|
+| **stateTransition**          | AbstractStateTransition/RawStateTransition | yes      |                                                         |
+
+Returns : {ValidationResult}
+
+## dpp.stateTransition.validateDate(stateTransition)
+
+**Description**: Validate State Transition Structure
```
Comment (thephez) [inline docs/usage/stateTransition.md:69]: Duplicate description?
```suggestion
**Description**: Validate State Transition Date
```

File: docs/usage/stateTransition.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+| parameters                   | type                      | required | Description                                             |  
+|------------------------------|---------------------------|----------| --------------------------------------------------------|
+| **stateTransition**          | AbstractStateTransition   | yes      |                                                         |
+
+Returns : {ValidationResult}
+
+## dpp.stateTransition.validateFee(stateTransition)
+
+**Description**: Validate State Transition Structure
```
Comment (thephez) [inline docs/usage/stateTransition.md:81]: Duplicate description?
```suggestion
**Description**: Validate State Transition Fee
```

File: docs/_sidebar.md (hunk 2 above)
Comment (Alex-Werner) [reply docs/_sidebar.md:5]: It was an issue indeed ! Thank you for noticing, fixed :)

**Review by thephez — APPROVED** [review]
:memo: Docs ACK (formatting, etc)

**Pull request merged**